### PR TITLE
Fixed using withTrashed and having an alias on a join clause

### LINKED ISSUE #96: useTableAliasInConditions breaks withTrashed / onlyTrashed
When trying to using withTrashed() I get errors that array_filter takes an array but gets a collection. 

I think the problem is when useTableAliasInConditions is called the wheres array on the PowerJoin class gets turned into a collection.

Hopefully adding a ->toArray() method before returning from the useTableAliasInConditions fixes the problem.

### PR DESCRIPTION
**Pull request by jonneroelofs**
fixes #96

### CODE REVIEW

**Comment by jonneroelofs** [comment]
One weird thing I noticed is that when you add an alias, as I did in the test:

```php
$join->as('myAlias');
```

That the test then actually works regardless of whether you actually add withTrashed(). So it seems to be the case that the alias accidentally (?) already removes the part that filters out the soft deleted records.

So if you comment withTrashed in the test I added on line 78, you will see that the test still works.

File: tests/SoftDeletesTest.php [hunk 1]
```diff
@@ -62,4 +62,31 @@ public function it_can_include_only_trashed()
...
         );
     }
+
+    /** @test */
+    public function it_can_disable_soft_deletes_when_using_an_alias()
+    {
+        // making sure the query doesn't fail
+        UserProfile::query()
+            ->joinRelationship('user', function ($join) {
+                $join->withTrashed();
```
Comment (luisdalmolin) [inline tests/SoftDeletesTest.php:72]: Was your intent to call `->as()` here too?

**Review by luisdalmolin — COMMENTED** [review]
This is looking good, I just left on question for you. Thanks for the PR, btw!

**Review by luisdalmolin — APPROVED** [review]

**Comment by luisdalmolin** [comment]
I published version 2.6.4 with this change. Thanks!

File: tests/SoftDeletesTest.php (hunk 1 above)
Comment (jonneroelofs) [reply tests/SoftDeletesTest.php:72]: I guess that would make more sense, to keep the test in line with the tother tests.

**Pull request merged**